libs/product/src/models/configurable-product.ts

```diff
@@ -3,17 +3,33 @@ import { DaffSortable } from '@daffodil/core';
 import { DaffProduct, DaffProductDiscount } from './product';
 import { DaffProductImage } from './product-image';
 
+/**
+ * A configurable product is a product with configurable attributes. The price of a configurable product may change based on 
+ * the attributes chosen, so a configurable product can have a price range. An example of a configurable product is a T-shirt.
+ */
 export interface DaffConfigurableProduct extends DaffProduct {
 	configurableAttributes: DaffConfigurableProductAttribute[];
 	variants: DaffConfigurableProductVariant[];
 }
 
+/**
+ * An attribute of the configurable product that the customer must choose to add the configurable product to the cart. 
+ * An example of an attribute would be size for clothing.
+ */
 export interface DaffConfigurableProductAttribute extends DaffSortable {
 	code:	string;
 	label: string;
 	values:	DaffConfigurableProductOptionValue[];
 }
 
+/**
+ * A variant is one version of the configurable product with all attributes chosen. Variants exist because you might not have
+ * all versions of every configuration of the product in stock. For example, you might have a medium, red shirt and a small,
+ * green shirt, but no small, red shirts. So you could have two variants (mediumRed, smallGreen) rather than the maximum 4 variants 
+ * (smallRed, mediumRed, smallGreen, mediumGreen). This ensures the customer can't add a configurable product to the cart that is not
+ * in stock. However, variants don't usually need to be considered by the frontend dev, because daffodil abstacts the concept of variants away into 
+ * an "available attributes" selector.
+ */
```

Comment:
I think we want to avoid speaking in the 2nd person. So instead of `Variants exist because you might not have all versions of every configuration of the product in stock` you would say `Variants exist because its possible that not all versions of every configuration of the product are in stock`.

FWIW I need to go back and fix this in my cart docs.

Reply:
Actually @damienwebdev what is your opinion on this? Is `you` and `your` okay or not?

Reply:
I think you're right. Probably better to go more formal with these

Reply:
changed